src/main/java/cubicchunks/worldgen/generator/flat/FlatTerrainProcessor.java

```diff
@@ -48,44 +48,33 @@
 @MethodsReturnNonnullByDefault
 public class FlatTerrainProcessor extends BasicCubeGenerator {
 
+    private final FlatGeneratorSettings conf;
+
     public FlatTerrainProcessor(ICubicWorld world) {
         super(world);
+        String json = world.getWorldInfo().getGeneratorOptions();
+        conf = FlatGeneratorSettings.fromJson(json);
     }
 
     @Override
     public ICubePrimer generateCube(int cubeX, int cubeY, int cubeZ) {
         ICubePrimer primer = new CubePrimer();
-
-        if (cubeY >= 0) {
-            return primer;
-        }
-        if (cubeY == -1) {
-            for (int x = 0; x < 16; x++) {
-                for (int z = 0; z < 16; z++) {
-                    primer.setBlockState(x, 15, z, Blocks.GRASS.getDefaultState());
-                    for (int y = 14; y >= 10; y--) {
-                        primer.setBlockState(x, y, z, Blocks.DIRT.getDefaultState());
-                    }
-                    for (int y = 9; y >= 0; y--) {
-                        primer.setBlockState(x, y, z, Blocks.STONE.getDefaultState());
-                    }
-                }
-            }
-            return primer;
-        }
-
-        for (int x = 0; x < 16; x++) {
-            for (int z = 0; z < 16; z++) {
-                for (int y = 0; y < 16; y++) {
-                    IBlockState state = Blocks.STONE.getDefaultState();
-                    if (Coords.localToBlock(cubeY, y) == world.getMinHeight()) {
-                        state = Blocks.BEDROCK.getDefaultState();
+        int floorY = cubeY * Cube.SIZE;
+        int topY = (cubeY + 1) * Cube.SIZE;
```

Comment:
Use Coords class instead of this. Coords.cubeToMinBlock and Coords.cubeToMaxBlock.